### PR TITLE
Get-DbaDatabaseState, speedups

### PR DESCRIPTION
**Pull request by niphlod**
## Type of Change

 - [ ] Bug fix (non-breaking change, fixes #<enter issue number>)
 - [ ] New feature (non-breaking change, adds functionality)
 - [ ] Breaking change (effects multiple commands or functionality)
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 - [x] Enhancement/speedup 

### Purpose
We already avoid enumerating via SMO for states. May as well use the same method to avoid enumerating databases.
Additionally, ScriptAnalyzer warning about hashtable has been silenced 'cause we _really_ wanna use a case-sensitive approach here.

### Approach
Reuse the state query to select databases by name, too

### Proof
```
     [+] Honors the Database parameter 674ms vs 8.89s
     [+] Honors the ExcludeDatabase parameter 4.76s vs 4.58s
     [+] Identifies online database 322ms vs 4.56s
     [+] Identifies offline database 283ms vs 4.46s
     [+] Identifies emergency database 324ms vs 4.53s
     [+] Identifies single_user database 4.34s vs 4.48s
     [+] Identifies restricted_user database 317ms vs 5.08s
     [+] Identifies multi_user database 317ms vs 4.43s
     [+] Identifies read_write database 291ms vs 4.5s
     [+] Identifies read_only database 309ms vs 4.5s
```
tl;dr : as good as it gets. Enumerating a single_user database via SMO seems unfortunately un-enhanceable

### CODE REVIEW

**Comment by potatoqualitee** [comment]
aweeeeeeeesome!
